packages/xo-server/src/xapi-stats.js

```diff
@@ -286,23 +272,24 @@ export default class XapiStats {
     return lastTimestamp
   }
 
-  _getStats(hostUuid, step, vmUuid) {
-    const hostStats = this._statsByObject[hostUuid][step]
+  _getCachedStats(hostUuid, step, vmUuid) {
```

Comment:
Only handle the stats of a single object here, no need to pass both a `hostUuid` and a `vmUuid`.

Reply:
In case of getting a Vm stats, i need also delete the host stats.

Reply:
Why?

Reply:
We fetch stats by host, if we delete Vm stats without deleting its host stats, we will considerate that we have the stats.